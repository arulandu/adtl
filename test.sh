#!/bin/bash
rm -f ./tests/*.out

for inf in ./tests/*.adtl; do
  name=$(basename "$inf" .adtl)
  outf="./tests/${name}.out"
  dune exec -- adtl -n -l "$inf" > "$outf"
  echo "Running... $inf -> $outf"
done
