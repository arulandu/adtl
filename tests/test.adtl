s : Type 0.
a : s.
e1 := fun x : Nat => s.
e2 := a. 
e3 := fun x : Nat => fun y : s => a.
e4 := succ 0.
e5 := elimNat e1 e2 e3 e4.
Context.
Eval e5.