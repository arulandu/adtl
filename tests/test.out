s : Type 0
a : s
e1 = fun _ : Nat => s
    : Nat -> Type 0
e2 = a
    : s
e3 = fun _ : Nat => fun _ : s => a
    : Nat -> s -> s
e4 = Succ 0
    : Nat
e5 = ElimNat (e1) (e2) (e3) (e4)
    : s
    = a
    : s
